get_defs.py

```diff
@@ -62,13 +67,10 @@ def extract_defs(word, vocab_type = None):
         print(f"No definitions of type {vocab_type} found for {word}")
     
     #Returns all definiitions of all types
-    elif api_call(word) != None:
-        for items in api_call(word):
+    elif current_call != None:
+        for items in current_call:
             for mean_items in items['meanings']:
                 for spec_defs in mean_items['definitions']:
                     all_defs.append(spec_defs['definition'])
         if len(all_defs) > 0:
             return all_defs
-    
-    return None
```

Comment:
1. consistency in return

2. naming can make this a lot more readable. Imagine something like "lookup_word" over api_call, and "result" over "current_call". Be careful of phrasing like "current_call" which imply the api call is ongoing.

The abstraction to a single api_call has reaped significant performance benefits!